src/IssueViz.Security/IssueVizSecurityPackage.cs

```diff
@@ -48,8 +49,13 @@ tool window.
     [PackageRegistration(UseManagedResourcesOnly = true, AllowsBackgroundLoading = true)]
     [Guid("D7D54E08-45E1-49A6-AA53-AF1CFAA6EBDC")]
     [ProvideMenuResource("Menus.ctmenu", 1)]
-    [ProvideToolWindow(typeof(HotspotsToolWindow), MultiInstances = false, Transient = true, Style = VsDockStyle.Tabbed, Window = VsWindowKindErrorList, Width = 700, Height = 250)]
+
+    [ProvideToolWindow(typeof(HotspotsToolWindow), MultiInstances = false, Transient = false, // Note: Transient must false when using ProvideToolWindowVisibility
+        Style = VsDockStyle.Tabbed, Window = VsWindowKindErrorList, Width = 700, Height = 250)]
+    [ProvideToolWindowVisibility(typeof(HotspotsToolWindow), LocalHotspotIssuesExistUIContext.GuidString)]
```

Comment:
Declare to VS that the visibility of the tool window is tied to the specific context.
Then we only need to set/clear the context and VS will do the rest.